test/test_script.py

```diff
@@ -1568,3 +1570,42 @@ def test_cmd_unique(self):
             self.parser.eval("$unique()")
         with self.assertRaisesRegex(ScriptError, areg):
             self.parser.eval("$unique(B:AB; D:C; E:D; A:A; C:X,1,:,extra)")
+
+    def test_cmd_countryname(self):
+        if not hasattr(builtins, 'gettext_countries'):
+            builtins.__dict__['gettext_countries'] = None
+        context = Metadata()
+        context["foo"] = "ca"
+        context["bar"] = ""
+        context["baz"] = "INVALID"
+
+        def mock_gettext_countries_en(arg):
+            return "Canada"
+
+        def mock_gettext_countries_ru(arg):
+            return "Канада"
```

Comment:
To fully mimic `gettext_countries()` those can be modified to:

```python
        def mock_gettext_countries_en(arg):
            return "Canada" if arg == 'Canada' else arg

        def mock_gettext_countries_ru(arg):
            return "Канада" if arg == 'Canada' else arg
```

If the string has not translation, it is returned as is (this is done after a conversion from country code to country name).

This would permit to test for unknown translation of a correct code:

```python
with mock.patch('builtins.gettext_countries', mock_gettext_countries_ru):
    self.assertScriptResultEquals("$countryname(fr,yes)", "France", context)
    self.assertScriptResultEquals("$countryname(INVALID,yes)", "", context)
```

And since English is the default language, the returned string is always the passed string, so we can simplify:

```python
        def mock_gettext_countries_en(arg):
            return arg
```

Reply:
Excellent suggestions.  Thank-you.